File: Sorting 1/selectionSort.java
//select minimum and then swap it
class Selection{
    public static void main(String[] args) {
        int[] arr = {13, 46, 24, 52, 20, 9};
        selectionSort(arr);
        for(int i=0;i<arr.length;i++){
            System.out.print(arr[i]+" ");
        }
    }
    public static void selectionSort(int[] arr){
        int n=arr.length;
        for(int i=0;i<=n-2;i++){
            int mini=i;
            for(int j=i;j<=n-1;j++){
                if(arr[j]<arr[mini]){
                    mini=j;
                }
            }
            int temp=arr[i];
            arr[i]=arr[mini];
            arr[mini]=temp;
        }
    }
}

File: Sorting 1/selectionSort.js
const selection =(arr)=>{
    let n=arr.length;
    for(let i=0;i<=n-2;i++){
        let mini=i;
        for(let j=i;j<=n-1;j++){
            if(arr[j]<arr[mini]){
                mini=j;
            }
        }
        [arr[i],arr[mini]]=[arr[mini],arr[i]];
    }
}
const arr = [64, 25, 12, 22, 11];
selection(arr);
console.log(arr);